pkg/retriever/retriever.go

```diff
@@ -158,69 +161,55 @@ func (retriever *Retriever) isAcceptableStorageProvider(storageProviderId peer.I
 // isAcceptableQueryResponse determines whether a queryResponse is acceptable
 // according to the current configuration. For now this is just checking whether
 // PaidRetrievals is set and not accepting paid retrievals if so.
-func (retriever *Retriever) isAcceptableQueryResponse(peer peer.ID, queryResponse *retrievalmarket.QueryResponse) bool {
+func (retriever *Retriever) isAcceptableQueryResponse(peer peer.ID, req types.RetrievalRequest, queryResponse *retrievalmarket.QueryResponse) bool {
 	// filter out paid retrievals if necessary
-	return retriever.config.PaidRetrievals || totalCost(queryResponse).Equals(big.Zero())
+
+	acceptable := retriever.config.PaidRetrievals || totalCost(queryResponse).Equals(big.Zero())
```

Comment:
```suggestion
	acceptable := retriever.config.PaidRetrievals || big.Add(big.Mul(queryResponse.MinPricePerByte, big.NewIntUnsigned(queryResponse.Size)), queryResponse.UnsealPrice).Equals(big.Zero())
```

I actually didn't mind this when you introduced it into the test so maybe we ditch `totalCost` entirely. It's not very hard to understand.